### PR TITLE
Fix SSH / SFTP confusion

### PR DESCRIPTION
**Pull request by ptesarik**
Recent versions of kdump recognize both ssh:// and sftp:// target URLs and treat them differently. Reflect this change in the UI.

### CODE REVIEW

**Comment by mvidner** [comment]
Thank you!

What do we mean by "recent versions of kdump"? In particular, is this PR suitable for inclusion in SLE12-SP1?

I have checked [kdump.changes](https://build.opensuse.org/package/view_file/Kernel:kdump/kdump/kdump.changes?expand=1&rev=4e82f19651ff9a900435352784e7cc7d) and it seems the relevant change is this one (even though it does not mention SFTP), which is in SLE12-GA.
```
Fri May 16 17:42:37 UTC 2014 - ptesarik@suse.cz

- Update to 0.8.12
  o Add "ssh" target protocol to transfer the file over the
    standard input to a remote host using the standard ssh client
    (bnc#868704).
```

**Comment by mvidner** [comment]
Also the [help text](https://github.com/ptesarik/yast-kdump/blob/b4c9302e802ad647483b9fa116bbe5544b0db9cb/src/include/kdump/helps.rb#L93) should be updated to mention SFTP. I have to admit that I don't know how I should choose between SSH and SFTP (and skimming [Wikipedia: SFTP](https://en.wikipedia.org/wiki/SSH_File_Transfer_Protocol) has not helped), so I would like the help text to mention that.

**Comment by ptesarik** [comment]
Ah, I completely missed the fact that there is a help text! ;-)
OK, I'll be glad to improve it.

**Comment by mvidner** [comment]
Please keep the PR open, we can add comments, commits, even rebase, until it is merged.

**Comment by mvidner** [comment]
Another thing: we have a CLI, so search for "ssh" in [clients/kdump.rb](https://github.com/ptesarik/yast-kdump/blob/b4c9302e802ad647483b9fa116bbe5544b0db9cb/src/clients/kdump.rb) and add "sftp" where appropriate. (Yes, the code is horribly repetitive.)

File: src/include/kdump/uifunctions.rb [hunk 1]
```diff
@@ -449,8 +451,8 @@ def tostringKDUMP_SAVE_TARGET
         end 
 
         # ssh
-      elsif Ops.get(@KDUMP_SAVE_TARGET, "target") == "ssh"
-        result = "ssh://"
+      elsif %w[ssh sftp].include?(Ops.get(@KDUMP_SAVE_TARGET, "target"))
```
Comment (mvidner) [inline src/include/kdump/uifunctions.rb]: Please use the regular array syntax, `["ssh", "sftp"]`, according to our [style guide](https://github.com/SUSE/style-guides/blob/master/Ruby.md#arrays).

**Comment by mvidner** [comment]
Last thing, let's formulate a .changes entry.

> - Dump targets:
>   - added the SFTP target.
>   - don't mention 'scp' as it was never used.

Do we want to mention some Bugzilla or Fate entry?

**Comment by kobliha** [comment]
Bugzilla or FATE Nr. is currently requested by BS guys ~ if we want to push it to SLE 12 SP1

File: src/include/kdump/uifunctions.rb [hunk 2]
```diff
@@ -570,10 +572,10 @@ def InitTargetKdump(key)
           Ops.get(@KDUMP_SAVE_TARGET, "server")
         )
         UI.ChangeWidget(Id("dir"), :Value, Ops.get(@KDUMP_SAVE_TARGET, "dir"))
-      elsif Ops.get(@KDUMP_SAVE_TARGET, "target") == "ssh"
+      elsif %w[ssh sftp].include?(Ops.get(@KDUMP_SAVE_TARGET, "target"))
         UI.ReplaceWidget(Id("Targets"), @ssh)
-        #UI::ChangeWidget(`id ("ssh"), `Value, true);
-        UI.ChangeWidget(Id("TargetKdump"), :Value, "ssh")
+        UI.ChangeWidget(Id("TargetKdump"), :Value,
+                        Ops.get(@KDUMP_SAVE_TARGET, "target"))
```
Comment (kobliha) [inline src/include/kdump/uifunctions.rb]: This is RYCP, please use `@KDUMP_SAVE_TARGET["target"]` instead

File: src/include/kdump/uifunctions.rb [hunk 3]
```diff
@@ -946,8 +948,8 @@ def StoreTargetKdumpHandle(type)
         else
           Ops.set(@KDUMP_SAVE_TARGET, "dir", "")
         end
-      elsif radiobut == "ssh"
-        Ops.set(@KDUMP_SAVE_TARGET, "target", "ssh")
+      elsif %w[ssh sftp].include?(radiobut)
+        Ops.set(@KDUMP_SAVE_TARGET, "target", radiobut)
```
Comment (kobliha) [inline src/include/kdump/uifunctions.rb]: Nitpicking: What's the difference between `radiobut` and `radiobutton`? This seems to be relic from the past, and should be consistent

File: src/include/kdump/uifunctions.rb [hunk 4]
```diff
@@ -449,8 +451,8 @@ def tostringKDUMP_SAVE_TARGET
         end 
 
         # ssh
-      elsif Ops.get(@KDUMP_SAVE_TARGET, "target") == "ssh"
-        result = "ssh://"
+      elsif %w[ssh sftp].include?(Ops.get(@KDUMP_SAVE_TARGET, "target"))
+        result = Ops.get(@KDUMP_SAVE_TARGET, "target") + "://"
```
Comment (kobliha) [inline src/include/kdump/uifunctions.rb]: just @KDUMP_SAVE_TARGET["target"]

File: src/include/kdump/uifunctions.rb [hunk 5]
```diff
@@ -321,7 +322,8 @@ def SetUpKDUMP_SAVE_TARGET(target)
           end
           # only ftp & ssh
           if Ops.get(@KDUMP_SAVE_TARGET, "target") == "ftp" ||
-              Ops.get(@KDUMP_SAVE_TARGET, "target") == "ssh"
+              Ops.get(@KDUMP_SAVE_TARGET, "target") == "ssh" ||
+              Ops.get(@KDUMP_SAVE_TARGET, "target") == "sftp"
```
Comment (kobliha) [inline src/include/kdump/uifunctions.rb]: Maybe `if ["ftp", "sftp", "ssh"].include?(@KDUMP_SAVE_TARGET["target"])` ?

**Comment by jreidinger** [comment]
@ptesarik ping

**Comment by ptesarik** [comment]
@jreidinger On my radar, but blocked. :-( I hope to have an updated request on Monday (May 25).

**Comment by ptesarik** [comment]
The new request tries to address all comments. More feedback welcome!

**Comment by mvidner** [comment]
Thanks, I have added commits of my own to make it Look Good To Myself, please check in #28.

**Comment by mvidner** [comment]
@ptesarik Merged, thank you!

**Pull request closed without merging**